Lib/Src/CThermistor.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file CThermistor.cpp
+ */
+
+/*
+ *  Created on: Feb 15, 2022
+ *      Author: niuslar
+ */
+
+#include "CThermistor.h"
+
+constexpr float CThermistor::s_calibration_coeff[] = {214, -166, 69.7, -13.4};
+constexpr uint8_t CThermistor::s_calibration_order = 3;
+
+/**
+ * @brief Constructor
+ * @param p_calibration_coeff Pointer to array with polynomial coefficients
+ * @param calibration_order Order of the polynomial equation
+ *
+ * @note The order of the polynomial is used in a strict mathematical sense,
+ i.e. third order polynomial will have 4 coefficients: a0, a1, a2, a3. The
+ minimum number of coefficients required is 2, which would make for a first
+ order polynomial otherwise knows as linear relation.
+ */
+CThermistor::CThermistor(const float *p_calibration_coeff,
+                         const uint8_t calibration_order)
+    : mp_calibration_coeff(p_calibration_coeff),
```

Comment:
Should copy contents into local member variable. All of the coefficients. we want for them to be local.